### ISSUE TITLE
new Object(argument) returns unexpected values

### ISSUE THREAD

**Issue by Toad06** [ruffle-rs/ruffle#1469]
`Object` without the `new` operator:

```
var a = Object("Object();");
var b = Object(2000);
var c = Object({c1:"hello",c2:"world"});
trace(a + "\n" + b + "\n" + c.c1 + "\n" + c.c2);
```

Same results in Flash and Ruffle:
> Object();
> 2000
> hello
> world

---

`Object` with the `new` operator:

```
var a = new Object("new Object();");
var b = new Object(2000);
var c = new Object({c1:"hello",c2:"world"});
trace(a + "\n" + b + "\n" + c.c1 + "\n" + c.c2);
```

Results in Flash:
> new Object();
> 2000
> hello
> world

Results in Ruffle:
> [object Object]
> [object Object]
> undefined
> undefined

Sample: [object_issue.zip](https://github.com/ruffle-rs/ruffle/files/5482852/object_issue.zip)

This bug affects #1464.

**Comment by Herschel** [comment]
Need to make `globals::object::constructor` act like `object::object_function`, but this is a little tricky because some of the primitive types need to turn into boxed `ValueObject` instead of a `ScriptObject`.

**Comment by Herschel** [comment]
Fixed by #2415.
